controllers/serviceclaim_controller.go

```diff
@@ -488,7 +488,50 @@ func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	genPred := predicate.GenerationChangedPredicate{}
+	reconcileOnRegisteredServiceUpdate := func(ctx context.Context, a client.Object) []reconcile.Request {
+		l := log.FromContext(ctx)
+		serviceclaims := &v1alpha1.ServiceClaimList{}
+		opts := &client.ListOptions{}
+		if err := r.List(ctx, serviceclaims, opts); err != nil {
+			return []reconcile.Request{}
+		}
+		rs, ok := a.(*v1alpha1.RegisteredService)
+		if !ok {
+			l.Info("error parsing object to RegisteredService when mapping to ServiceClaim reconciliation trigger", "object", a)
+			return []reconcile.Request{}
+		}
+		reply := make([]reconcile.Request, 0, len(serviceclaims.Items))
+		for _, sc := range serviceclaims.Items {
+			env := sc.Spec.EnvironmentTag
```

Comment:
can we just use the `status.registeredService` field to decide if we need to reconcile the ServiceClaim?

Reply:
Yeah by doing taht we can avoid the sci matching being duplicated